design/scheduler.md

```diff
@@ -0,0 +1,158 @@
+# Scheduler design
+
+This document covers the design and implementation details of the swarmkit
+scheduler.
+
+## Overview
+
+The scheduler becomes responsible for tasks once they reach the `Pending` state.
```

Comment:
it would be helpful to give more context to the `Pending` state.  I'm not sure what would read best, but something short about how a task reaches the state would better explain where the scheduler lives in the control flow.